### ISSUE TITLE
[Feature request] Add an exit code which disables the automatic restart of the adapter instance

### ISSUE THREAD

**Issue by AlCalzone** [ioBroker/ioBroker.js-controller#197]
A potential application is the following:
The adapter detects that a dependency is missing or the user has to do something else before starting the adapter, e.g. the newest tradfri version:
```TS
try {
	// tslint:disable-next-line:no-var-requires
	require("node-tradfri-client");
} catch (e) {
	console.error(`The module "node-aead-crypto" was not installed correctly!`);
	console.error(`To try reinstalling it, goto "${path.join(__dirname, "..")}" and run`);
	console.error(`npm install --production`);
	console.error(`If that fails due to missing access rights, run`);
	console.error(`${process.platform !== "win32" ? "sudo " : ""}npm install --production --unsafe-perm`);
	console.error(`instead. Afterwards, restart this adapter.`);
	process.exit(1);
}
```

However, calling `process.exit(1)` causes the js-controller to restart the instance after a while, thus resulting in a loop.

It would be nice to have an exit code that tells the controller to stop restarting the instance until the user starts it again.

**Comment by GermanBluefox** [comment]
adapter.extendForeignObject('system.adapter.' + adapter.namespace, {common:
{enabled: false}});

??

2018-05-14 21:52 GMT+02:00 AlCalzone <notifications@github.com>:

**Comment by AlCalzone** [comment]
That is one way, but not possible in my case. The problem happens before the adapter instance is created, so I only really have `process.exit`.
